tycho-bundles/org.eclipse.tycho.p2.resolver.impl.test/src/test/java/org/eclipse/tycho/p2/resolver/P2ResolverTest.java

```diff
@@ -506,11 +509,26 @@ public void testResolveWithoutProject() throws Exception {
     }
 
     private P2TargetPlatform getTargetPlatform() {
-        return tpFactory.createTargetPlatform(tpConfig, NOOP_EE_RESOLUTION_HANDLER, reactorProjects, pomDependencies);
+        return getTargetPlatform(NOOP_EE_RESOLUTION_HANDLER);
+    }
+
+    private P2TargetPlatform getTargetPlatform(boolean finalTarget) {
+        return getTargetPlatform(NOOP_EE_RESOLUTION_HANDLER, finalTarget);
```

Comment:
I agree that the term TargetPlatform, FinalTargetPlatform an PreliminaryTargetPlatform and TargetPlatformFile are all a bit confusingly used inside tycho.
In this partifular case it seems the unitest sometimes can only work on a PreliminaryTargetPlatform because the test does not setup all things for a FinalTargetPlatform ...

Currently there is a PreliminaryTargetPlatform that is used to read all dependencies, projects are referenced here for the build order and then in the TargetPlatformMojo this is made "final" for a given project. Beside That there is a "Target" that is the collection of items from the repository and target files ...

Given the wording in PDE, what tycho is calling a TargetPlatform would better map to the "Plugin-Dependencies-Container"  so PluginDependencies might be a better term than TargetPlatform and then PluginDependencies could be distinguished between "computed" and "resolved" versus Preliminary/Final ...